x-pack/plugins/alerting/server/alerts_client.ts

```diff
@@ -0,0 +1,248 @@
+    this.alertTypeRegistry = alertTypeRegistry;
+    this.savedObjectsClient = savedObjectsClient;
+  }
+
+  public async create({ data, options }: CreateOptions) {
+    // Throws an error if alert type isn't registered
+    const alertType = this.alertTypeRegistry.get(data.alertTypeId);
+    const validatedAlertTypeParams = validateAlertTypeParams(alertType, data.alertTypeParams);
+    const { alert: rawAlert, references } = this.getRawAlert({
+      ...data,
+      alertTypeParams: validatedAlertTypeParams,
+    });
+    const createdAlert = await this.savedObjectsClient.create('alert', rawAlert, {
+      ...options,
+      references,
+    });
+    let scheduledTask;
+    try {
+      scheduledTask = await this.scheduleAlert(createdAlert.id, rawAlert, this.basePath);
+    } catch (e) {
+      // Cleanup data, something went wrong scheduling the task
+      try {
+        await this.savedObjectsClient.delete('alert', createdAlert.id);
+      } catch (err) {
+        // Skip the cleanup error and throw the task manager error to avoid confusion
+        this.log(
+          ['alerting', 'error'],
+          `Failed to cleanup alert "${createdAlert.id}" after scheduling task failed. Error: ${
+            err.message
+          }`
+        );
+      }
+      throw e;
+    }
+    await this.savedObjectsClient.update(
+      'alert',
+      createdAlert.id,
+      {
+        scheduledTaskId: scheduledTask.id,
+      },
+      { references }
+    );
+    createdAlert.attributes.scheduledTaskId = scheduledTask.id;
+    return this.getAlertFromRaw(createdAlert.id, createdAlert.attributes, references);
+  }
+
+  public async get({ id }: { id: string }) {
+    const result = await this.savedObjectsClient.get('alert', id);
+    return this.getAlertFromRaw(result.id, result.attributes, result.references);
+  }
+
+  public async find({ options = {} }: FindOptions = {}) {
+    const results = await this.savedObjectsClient.find({
+      ...options,
+      type: 'alert',
+    });
+    return results.saved_objects.map(result =>
+      this.getAlertFromRaw(result.id, result.attributes, result.references)
+    );
+  }
+
+  public async delete({ id }: { id: string }) {
+    const alertSavedObject = await this.savedObjectsClient.get('alert', id);
+    const removeResult = await this.savedObjectsClient.delete('alert', id);
+    await this.taskManager.remove(alertSavedObject.attributes.scheduledTaskId);
+    return removeResult;
+  }
+
+  public async update({ id, data, options = {} }: UpdateOptions) {
+    const existingObject = await this.savedObjectsClient.get('alert', id);
+    const { alertTypeId } = existingObject.attributes;
+    const alertType = this.alertTypeRegistry.get(alertTypeId);
+
+    // Validate
+    const validatedAlertTypeParams = validateAlertTypeParams(alertType, data.alertTypeParams);
+
+    const { actions, references } = this.extractReferences(data.actions);
+    const updatedObject = await this.savedObjectsClient.update(
```

Comment:
Correct for now, after the next execution, when it gets to set the next schedule time, that's when it will take in the updated interval. We're looking at changing this soon.